### ISSUE TITLE
GCM Authorization Failed

### ISSUE THREAD

**Issue by egarim** [Redth/PushSharp#574]
I got "GCM Authorization Failed" , when I try to send push notifications to android, here is the constructor I'm using for the gcm configuration:

GcmConfiguration newGcmConfiguration = new GcmConfiguration("app id", "api key","testPush.TestPush");

**Comment by Redth** [comment]
Try just using the API Key overload of the ctor.

**Comment by egarim** [comment]
Hi Redth, here is the snippet I'm using, if I only use the api key constructor I get a 400 error bad request

GcmConfiguration newGcmConfiguration = new GcmConfiguration("my api key here");


            GcmServiceBroker Broker = new GcmServiceBroker(newGcmConfiguration);

            // Wire up events
            Broker.OnNotificationFailed += (notification, exception) =>
            {
                Console.WriteLine("Notification Failed!");
            };
            Broker.OnNotificationSucceeded += (notification) =>
            {
                Console.WriteLine("Notification Sent!");
            };

            // Start the broker
            Broker.Start();

           

            GcmNotification newGcmNotification = new GcmNotification();
            newGcmNotification.Data =JObject.Parse("{\"alert\":\"Hello World!\",\"badge\":7,\"sound\":\"sound.caf\"}");
            Broker.QueueNotification(newGcmNotification);

            Console.ReadLine();
            // Stop the broker, wait for it to finish   
            Broker.Stop();

**Comment by dsimon-ac** [comment]
Hey there, I got some 401 Authorization Failed too when I was sending more than one notification. The reason Google returns 401 was that my request contains many authorization keys in the header field.

In my opinion the relevant position in code is here:
https://github.com/Redth/PushSharp/blob/3.0-dev/PushSharp.Google/GcmServiceConnection.cs#L55

When I add something like this it works for me:
```c#
if (http.DefaultRequestHeaders.Contains("Authorization"))
	http.DefaultRequestHeaders.Remove("Authorization");

http.DefaultRequestHeaders.TryAddWithoutValidation("Authorization", "key=" + Configuration.SenderAuthToken);
```

**Comment by egarim** [comment]
Hey Daniel can you share a sample code?

**Comment by Ska-K** [comment]
This worked for me. Thanks @dsimon-ac !

**Comment by DanDude0** [comment]
I have submitted a fix for this issue, pull request #582

**Comment by Redth** [comment]
Merged #582 which looks like a reasonable fix.  Thanks @DanDude0 for helping!
